### DISCUSSION TITLE
vagrant and fedora 36 (beta)

### DISCUSSION THREAD

**Discussion by buckaroogeek** [vagrant-libvirt/vagrant-libvirt#1478] [Show and tell]
If you use vagrant (upstream) on Fedora 35 along with vagrant-libvirt installed as a plugin (and not rpm) you might be interested in the info below.

I updated my F35 workstation to F36 beta. The F35 workstation had vagrant 2.2.19 installed from upstream (vagrant.x86_64.rpm) along with vagrant-libvirt installed as a plugin using the excellent information in the vagrant-libvirt readme.  The update to F36 beta was successful and without issue. However I noticed that during the upgrade process the vagrant 2.2.19 rpm from the fedora repo was installed along with the vagrant-libvirt plugin. 

Both vagrant up and vagrant plugin update failed with errors. I removed the vagrant-2.2.19-4.fc36.noarch rpm using dnf which also removed a number of dependencies including the vagrant-libvirt rpm.  I installed vagrant from upstream using the following steps (modified from Hashicorp):
1. sudo dnf install -y dnf-plugins-core
2. sudo dnf config-manager --add-repo https://rpm.releases.hashicorp.com/fedora/hashicorp.repo
3. sudo dnf -y install vagrant.x86_64

Step 3 above has the modified instruction (i.e. set the x86_64 arch) which will select vagrant from the hashicorp repo.

I the vagrant-libvirt plugin was already present (hold over from F35) and current. vagrant plugin update worked fine but was not needed. vagrant up worked without issue.

regards

**Comment by electrofelix** [comment]
If you want to switch to the distro packaged vagrant rather than the upstream one for fedora 36, you should have success with the following approach as well (avoids the linking issues needing to patch the upstream vagrant libraries assuming that is still needed).

```
dependencies=$(sudo dnf repoquery --qf "%{name}" $(for dep in $(sudo dnf repoquery --depends vagrant-libvirt 2>/dev/null | cut -d' ' -f1); do echo "--whatprovides ${dep} "; done) 2>/dev/null)
sudo dnf install @virtualization ${dependencies}
sudo dnf remove vagrant-libvirt
sudo systemctl enable --now libvirtd
vagrant plugin install vagrant-libvirt
```

It's what I've been recommending to a few people who've encountered issues. Hopefully will get to rework the readme to make it easier to have the different installation approaches covered better in the future.

**Comment by buckaroogeek** [comment]
One additional note, as Scott Willams notes ([https://ask.fedoraproject.org/t/vagrant-and-fedora-36-beta/21335]) future runs of dnf update may pull the fedora rpms of vagrant and replace the upstream version. If you want to retain upstream vagrant then you can follow Scott's recommendation and append exclude=vagrant* to the fedora repository configuration files in /etc/yum.repo.d. 

If you also use the dnf local plugin (recommended when also using vagrant to host fedora vms) then the dnf update command would look like: `sudo dnf update -x vagrant*`

If you switch from upstream to the fedora vagrant package, then I would also recommend deleting or moving aside ~/.vagrant before installing the fedora package. This creates a clean environment for the different vagrant package.
